exercise-integration-testing-5-53/test/student-submissions/yoav-ben-zvi.test.js

```diff
@@ -0,0 +1,186 @@
+        //Arrange
+        const eventToAdd = {
+            temperature: 20,
+            category: 'parents-room',
+            name: "Event-querying",
+            color: "Green",
+            weight: "80 gram",
+            status: "active",
+        };
+
+        //Act
+        await request(expressApp).post("/sensor-events").send(eventToAdd);
+        const getEventsResponse = await request(expressApp).get("/sensor-events/parents-room/name").send(); 
+
+        //Assert
+        expect(getEventsResponse.body[0]).toMatchObject({...eventToAdd});
+        expect(getEventsResponse.body).toHaveLength(1);
+    });
+    // Mhhmmm, i feel like i'm either missing something here or more likely, i didn't get the excercise properly
+
+    // ✅ TASK: Test that querying for /sensor-events route works when there are multiple events
+    test("When querying for /sensor-events when there are multiple events, route should work", async () => {
+        //Arrange
+        const eventToAdd = {
+            temperature: 20,
+            category: 'storage-room',
+            name: "Event-querying-1",
+            color: "Green",
+            weight: "80 gram",
+            status: "active",
+        };
+
+        const eventToAddTwo = {
+            temperature: 20,
+            category: 'storage-room',
+            name: "Event-querying-2",
+            color: "Green",
+            weight: "80 gram",
+            status: "active",
+        };
+
+        const eventToAddThree = {
+            temperature: 20,
+            category: 'storage-room',
+            name: "Event-querying-3",
+            color: "Green",
+            weight: "80 gram",
+            status: "active",
+        };
+
+        //Act
+        const returnedEvent = await request(expressApp).post("/sensor-events").send(eventToAdd);
+        const returnedEventTwo = await request(expressApp).post("/sensor-events").send(eventToAddTwo);
+        const returnedEventThree = await request(expressApp).post("/sensor-events").send(eventToAddThree);
+        const getEventsResponse = await request(expressApp).get("/sensor-events/storage-room/name").send(); 
+        
+        //Assert
+        expect(getEventsResponse.body).toEqual([
+            returnedEvent.body,
+            returnedEventTwo.body,
+            returnedEventThree.body,
+        ])
+        expect(getEventsResponse.body).toHaveLength(3);
+    });
+
+    // ✅ TASK: Test that querying for /sensor-events route and sorting by the field 'name', the results are indeed sorted
+    test("When querying for /sensor-events and sorting by 'name' key, should return a sorted array according to the event's name", async () => {
+        //Arrange
+        const eventToAdd = {
+            temperature: 20,
+            category: 'play-room',
+            name: "B",
+            color: "Green",
+            weight: "80 gram",
+            status: "active",
+        };
+
+        const eventToAddTwo = {
```

Comment:
Nitty-gritty - Call this firstEventToReturn, or AEvent - something that makes it clear what is the sort order

In a way, tests are about minimizing the reader brain CPU cycles 😅